gateway/src/apicast/oauth/oidc.lua

```diff
@@ -194,8 +196,17 @@ function _M:transform_credentials(credentials, cache_key)
   end
 
   local payload = jwt_obj.payload
-
   local app_id = payload.azp
+
+  if self.service and self.service.oidc and self.service.oidc.client_id then
+    local tmpl, tmpl_err = TemplateString.new(self.service.oidc.client_id, "liquid")
```

Comment:
We could create the `TemplateString` in the constructor to avoid doing this in every request.